graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/processor/FieldModelBuilder.java

```diff
@@ -319,9 +319,15 @@ private static boolean isCompatible(Column.ColumnType type1, Column.ColumnType t
         throw SkipException.INSTANCE;
       }
     }
-    return cqlIndexDirective.map(
-        d ->
-            new IndexModelBuilder(d, parentCqlName, cqlName, cqlType, messagePrefix, context)
-                .build());
+    return cqlIndexDirective.flatMap(
```

Comment:
Why not move this up into the if statement above?

```java
if (cqlIndexDirective.isPresent()) {
   // ...
   return Optional.of(
     new IndexModelBuilder(d, parentCqlName, cqlName, cqlType, messagePrefix, context)
        .build());
}
```

Reply:
Or move the other checks into the `flatMap()`? 

```java
return cqlIndexDirective.flatMap(
        d -> {
          try {

            if (isPk) {
              invalidMapping("%s: partition or clustering columns can't have an index", messagePrefix);
              throw SkipException.INSTANCE;
            }
            if (isUdtField) {
              invalidMapping("%s: UDT fields can't have an index", messagePrefix);
              throw SkipException.INSTANCE;
            }
            return Optional.of(
                new IndexModelBuilder(d, parentCqlName, cqlName, cqlType, messagePrefix, context)
                    .build());
          } catch (SkipException e) {
            return Optional.empty();
          }
        });
```

Reply:
👍 I applied your first suggestion.